xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -588,6 +607,16 @@ where
         // and roll the transaction back, so we can fetch updates from the server before
         // being ready to process the message for a second time.
         let mut processed_message = None;
+        if is_test_mode_aead_msg() {
+            return Err(OpenMlsProcessMessage(ProcessMessageError::ValidationError(
+                ValidationError::UnableToDecrypt(AeadError),
+            )));
+        }
+
+        if is_test_mode_future_wrong_epoch() {
+            //this is just a mock for testing, the whole function only appears in tests
```

Comment:
I'm not sure where we landed on the mocking - are we keeping this forever, or do we have an idea for the future that's just too costly right now?

Reply:
here is the issue to use a mocking tool later and refactor the code to make it more mocking-friendly, atm it requires lot of refactoring.
https://github.com/xmtp/libxmtp/issues/1903